pages/admin/challenge/choose-opponent.js

```diff
@@ -1,18 +1,15 @@
 import React from 'react';
 import { components } from '../../../components/challenge';
-import Layout from '../../../components/layout';
 import { constants } from '../../../components/common';
+import withAuthSync from '../../../hocs/withAuthSync';
 
-const { SELECTED_MENU_ITEM, NEXT_LINKS } = constants;
-const { challenge } = SELECTED_MENU_ITEM;
+const { NEXT_LINKS } = constants;
 const { challengeInfoLink } = NEXT_LINKS;
 
 const { OpponentSelect } = components;
 
 const ChooseOpponentPage = () => (
-    <Layout selectedMenuItem={challenge}>
-        <OpponentSelect next={challengeInfoLink} />
-    </Layout>
+    <OpponentSelect next={challengeInfoLink} />
 );
 
-export default ChooseOpponentPage;
+export default withAuthSync(ChooseOpponentPage);
```

Comment:
i still dont completely understand what you are wrapping these component in the `withAuthSync` HOC

Reply:
the HOC is basically for protected routes. If a user isn't logged in or his session has expired, it redirects the user to the login page

Reply:
Why isn't there a piece of data in the redux state that says that a user is logged in?

Reply:
I'm using the session data to check that